### PR TITLE
Add upgrade downgrade tests for reparent

### PR DESCRIPTION
**Pull request by GuptaManan100**
## Description

Continuation of #9300. Follows the same methodology as described there. This PR adds a workflow for the upgrade-downgrade tests for reparent operations.

## Related Issue(s)

- #7344 

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

File: go/test/endtoend/reparent/newfeaturetest/reparent_test.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
+/*
+Copyright 2020 The Vitess Authors.
+
```
Comment (frouioui) [inline go/test/endtoend/reparent/newfeaturetest/reparent_test.go:3]: Not sure if this file should in a `incompatible` package instead of `newfeature`

**Review by frouioui — APPROVED** [review]
LGTM 🚀

File: go/test/endtoend/reparent/newfeaturetest/reparent_test.go (hunk 1 above)
Comment (GuptaManan100) [reply go/test/endtoend/reparent/newfeaturetest/reparent_test.go:3]: What I meant by `newfeature` was that these are the tests for the changes or new features that have gone into the reparent functions which will only work after all the components are upgraded. Basically, after release 13, all the tests in `newfeature` will move to the normal tests and new ones will go here.

Comment (deepthi) [reply go/test/endtoend/reparent/newfeaturetest/reparent_test.go:3]: header should be 2022 😀

Comment (GuptaManan100) [reply go/test/endtoend/reparent/newfeaturetest/reparent_test.go:3]: Living in denial ft. me 😆

Comment (GuptaManan100) [reply go/test/endtoend/reparent/newfeaturetest/reparent_test.go:3]: I'll fix it

**Review by deepthi — APPROVED** [review]
Nice work! 💯

**Pull request merged**